src/sonic-dhcp-server/dhcp_server/common/dhcp_db_monitor.py

```diff
@@ -1,131 +1,419 @@
-                        need_refresh = True
-            # For del operation, we can skip disabled change
-            if op == "DEL":
-                if key in enabled_dhcp_interfaces:
-                    need_refresh = True
-        return need_refresh
+        for table in checker_names:
+            if table not in self.checker_dict:
+                syslog.syslog(syslog.LOG_ERR, "Cannot find checker for {} in checker_dict".format(table))
+                continue
+            self.checker_dict[table].disable()
 
-    def _check_vlan_update(self, enabled_dhcp_interfaces):
+    def enable_checkers(self, checker_names):
         """
-        Check vlan table
+        Enable checkers
         Args:
-            enabled_dhcp_interfaces: DHCP interface that enabled dhcp_server
-        Returns:
-            Whether need to refresh
+            checker_names: set contains name of tables need to be enable
         """
-        need_refresh = False
-        while self.subscribe_vlan_table.hasData():
-            key, op, _ = self.subscribe_vlan_table.pop()
-            # For vlan doesn't have related dhcp entry, not need to refresh dhcrelay process
-            if key not in enabled_dhcp_interfaces:
+        for table in checker_names:
+            if table not in self.checker_dict:
+                syslog.syslog(syslog.LOG_ERR, "Cannot find checker for {} in checker_dict".format(table))
                 continue
-            need_refresh = True
-        return need_refresh
+            self.checker_dict[table].enable(self.db_connector.config_db)
 
-    def _check_vlan_intf_update(self, enabled_dhcp_interfaces):
+    def check_db_update(self, db_snapshot):
         """
-        Check vlan_interface table
+        Fetch db and check update
         Args:
-            enabled_dhcp_interfaces: DHCP interface that enabled dhcp_server
+            db_snapshot: dict contains db snapshot parameter
         Returns:
-            Whether need to refresh
+            Whether need to refresh config file for kea-dhcp-server
         """
+        state, _ = self.sel.select(self.select_timeout)
+        if state == swsscommon.Select.TIMEOUT or state != swsscommon.Select.OBJECT:
+            return False
         need_refresh = False
-        while self.subscribe_vlan_intf_table.hasData():
-            key, _, _ = self.subscribe_vlan_intf_table.pop()
-            splits = key.split("|")
-            vlan_name = splits[0]
-            ip_address = splits[1].split("/")[0] if len(splits) > 1 else None
-            if vlan_name not in enabled_dhcp_interfaces:
-                continue
-            if ip_address is None or ipaddress.ip_address(ip_address).version != 4:
+        for checker in self.checker_dict.values():
+            if not checker.is_enabled():
                 continue
-            need_refresh = True
+            need_refresh |= checker.check_update_event(db_snapshot)
```

Comment:
If of checker returns True, then stop `check_update_event` for rest checkers, just invoke clear_events for them and return True.

Reply:
Fixed